x-pack/plugins/observability_solution/synthetics/server/routes/monitor_cruds/add_monitor/utils.ts

```diff
@@ -42,41 +42,33 @@ export const parseMonitorLocations = (
     pvtLocs = prevLocations.filter((loc) => !loc.isServiceManaged).map((loc) => loc.id);
   } else {
     if (prevLocations && !internal) {
+      const prevPublicLocs = prevLocations
+        .filter((loc) => loc.isServiceManaged)
+        .map((loc) => loc.id);
+      const prevPrivateLocs = prevLocations
+        .filter((loc) => !loc.isServiceManaged)
+        .map((loc) => loc.id);
```

Comment:
Not a big deal at all but we could shorten this from O(2n) to O(n) by using `reduce` and bucketing the locations into separate keys on a single object or array.

Pseudocode:

```typescript
// could reduce to an object with two keys as well
const [prevPublicLocs, prevPrivateLocs] = prevLocations.reduce((prev, { id, isServiceManaged }) => {
  if (isSeviceManaged) return [[...prev[0], id], prev[1]];
  return [prev[0], [...prev[1], id]];
, [[], []]);
```